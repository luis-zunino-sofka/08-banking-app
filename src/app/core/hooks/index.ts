export * from "./signup";
export * from "./login";
export * from "./dashboard";
export * from "./withdraw";
export * from "./purchase";
export * from "./appContainer";
export * from "./accountProvider";
export * from "./navbar";
