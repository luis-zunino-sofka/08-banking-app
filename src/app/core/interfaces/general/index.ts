export * from "./errors.interface";
export * from "./dinHeader.interface";
export * from "./navItem.interface";
