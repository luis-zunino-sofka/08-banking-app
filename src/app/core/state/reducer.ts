import { IAction, IState } from "@core/interfaces";
import { loginCases, loginInitialState } from "./login";
import { globalBalanceCases, globalBalanceInitialState } from "./globalBalance";

export const initialState = {
  ...loginInitialState,
  ...globalBalanceInitialState,
};

export const reducer = (state: IState, action: IAction): IState => {
  const cases = {
    ...loginCases,
    ...globalBalanceCases,
  };
  if (action.type && cases[action.type]) {
    return cases[action.type](state, action.payload);
  }
  return state;
};
