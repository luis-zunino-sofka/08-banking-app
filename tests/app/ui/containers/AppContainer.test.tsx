import React, { ReactNode } from "react";
import {
  AccountContext,
  IAccountContext,
} from "../../../../src/app/core/state/context";
import { describe, expect, test, vi } from "vitest";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { AppContext } from "../../../../src/app/core/state/AppContext";
import { IState } from "../../../../src/app/core/interfaces";
import { mockAppState, mockAccountState } from "../../../mocks/dataMocked";

interface IProps {
  children: ReactNode;
  mockAppState: IState;
  mockAccountState: IAccountContext;
}
export const appWrapper = ({
  children,
  mockAppState,
  mockAccountState,
}: IProps) => (
  <AppContext.Provider value={mockAppState}>
    <AccountContext.Provider value={mockAccountState}>
      {children}
    </AccountContext.Provider>
  </AppContext.Provider>
);

export const AccountContextMock = ({
  children,
  mockAppState,
  mockAccountState,
}: IProps) => appWrapper({ children, mockAppState, mockAccountState });

vi.mock("../../../../src/app/core/state/AppContext");
describe("AccountProvider", () => {
  test("Match snapshoot", () => {
    const { asFragment } = render(
      <AppContext.Provider value={mockAppState}>
        <AccountContext.Provider value={mockAccountState}>
          <></>
        </AccountContext.Provider>
      </AppContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
