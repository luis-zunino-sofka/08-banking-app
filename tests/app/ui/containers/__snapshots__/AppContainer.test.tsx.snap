// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountProvider > Match snapshoot 1`] = `<DocumentFragment />`;
